### DISCUSSION TITLE
github push command sometimes waits for hours

### DISCUSSION THREAD

**Discussion by adriatic** [community/community#15222] [Other Feature Feedback, Questions, & Ideas]
Here is the example where after waiting for good 10 minutes, I interrupted the first `push` and send the same request again (as it helps sometimes). This time did not help and I am still waiting.

```
nik$ git push                           
Enumerating objects: 23, done.
Counting objects: 100% (23/23), done.
Delta compression using up to 6 threads
Compressing objects: 100% (11/11), done.
Writing objects: 100% (13/13), 6.23 MiB | 1.14 MiB/s, done.
Total 13 (delta 7), reused 0 (delta 0), pack-reused 0
^C
nik$ git push
Enumerating objects: 23, done.
Counting objects: 100% (23/23), done.
Delta compression using up to 6 threads
Compressing objects: 100% (11/11), done.
Writing objects: 100% (13/13), 6.23 MiB | 1.29 MiB/s, done.
Total 13 (delta 7), reused 0 (delta 0), pack-reused 0
```
Using github desktop to manage this repo results with same behavior

<img width="721" alt="image" src="https://user-images.githubusercontent.com/2712405/164086054-e785e525-bb10-4393-815e-f4fc32fb3281.png">

**Comment by Odabasyusuf** [comment]
Hi! Did you find a solution?

**Reply by adriatic** [reply]
While I had initially no problem in several weeks, I had to delete the connection between Github and Netlify. After that everything is fine again. That however is not a solution 😸

**Reply by ashish-kus** [reply]
Total 103 (delta 10), reused 0 (delta 0), pack-reused 0
  
I am stuck at this and never used Netlify.

**Reply by Lugonathaly** [reply]
Hola quiero saber cómo hace mi propia licencia en github y si eso se puede hace

**Comment by Abbasi233** [comment]
same issue

**Comment by dcsuka** [comment]
Same issue for me as well.

EDIT: I am running a Mac M1 OS.

**Comment by f1ght3rlucky** [comment]
same issue

**Reply by dcsuka** [reply]
For me, it seems to be the case that it happens primarily with large pushes. If I make the pushes smaller, it works better. Is that so for you as well?

**Reply by f1ght3rlucky** [reply]
**git gc** 

use this command and it will be fixed

**Reply by Jeffery18-hub** [reply]
cool man, this works.  THX

**Reply by kemalsanli** [reply]
+1 working fine after

**Reply by Pachimanok** [reply]
working for me!!

**Reply by raoulelysee** [reply]
Thanks!

**Reply by jnroot0101** [reply]
Thank you man!!

**Reply by AmazinglyStrange** [reply]
Are you a wizard?  Worked like magic!

**Comment by qadir-genetech** [comment]
me too getting this problem

**Reply by dcsuka** [reply]
This only happens to me when the commits are too large, or I do many at once. Try committing only one hash at a time.

**Comment by RISMALUKMANHAKIM** [comment]
Enumerating objects: 2615, done. Counting objects: 100% (2615/2615), done. Delta compression using up to 4 threads Compressing objects: 100% (2466/2466), done. Writing objects: 100% (2615/2615), 25.58 MiB | 5.33 MiB/s, done. error: RPC failed; curl 18 transfer closed with outstanding read data remaining send-pack: unexpected disconnect while reading sideband packet fatal: the remote end hung up unexpectedly Everything up-to-date

**Reply by georgialive** [reply]
try: `git gc` in your terminal

**Reply by 01119257547aass** [reply]
مكالمات ورسايل ايمو وواتساب فيديو

**Comment by channyeintun** [comment]
`git gc` doesn't help.

**Reply by MohammedMogeab** [reply]
it helps more that you think . just make `git gc ` after you have stuck end it and write that commend . the problem will be solve .

**Comment by Chyba1997** [comment]
@Chyba1997

**Comment by Chyba1997** [comment]
@Chyba1997 gogo twitters

**Reply by Chyba1997** [reply]
![image](https://user-images.githubusercontent.com/131381083/236300930-4da8c8fb-817a-4638-b34d-b26bf9ad487b.png) https://twitter.com/maksim_chyba/status/1654197118750412826?s=12

**Comment by Chyba1997** [comment]
- https://twitter.com/maksim_chyba/status/1654197118750412826?s=12

**Comment by Chyba1997** [comment]
https://play.rpsleague.io/BnB

**Comment by Chyba1997** [comment]
https://play.rpsleague.io/BnB @Chyba1997 [****](https://play.rpsleague.io/)

**Comment by ospaarmann** [comment]
Same issue for me. The complete push was 2.71 MiB. Git gets stuck on GitHub Desktop and on the command line. Sometimes a restart helps. This happens more and more often and really needs to be fixed. 

```bash
🤷 ❯ git push
Enumerating objects: 30, done.
Counting objects: 100% (30/30), done.
Delta compression using up to 10 threads
Compressing objects: 100% (24/24), done.
Writing objects: 100% (27/27), 2.72 MiB | 491.00 KiB/s, done.
Total 27 (delta 11), reused 0 (delta 0), pack-reused 0
```

I'm on a MacBook Pro, 14-inch, 2021, Apple M1 Max, 64 GB RAM.

**Reply by TimmyBest** [reply]
happens all the time, something is really wrong!!

**Comment by mehmetyilmaz001** [comment]
There was a gif file in the .github folder in my case. When I delete that file everything went ok.

**Comment by adriatic** [comment]
I started this thread more than a year ago and had problems since I renewed the connection between GitHub and Netlify. 

Note that the common factor in all the described problems is the failure of the (distributed) action initiated by the command “git push” issued by the user via the terminal shell. As stated above, this command failed to execute, because Netlify did not recognize my account on GitHub that tried to upload my app’s sources to Netlify for deployment.

This scenario is not the only one that can cause the command “git push” to fail - since this thread started, GitHub requested 2FA authentication, making it necessary to ensure that your terminal shell is aware of that. This applies to both ways sending the push command (using the terminal shell or GitHub desktop application.

**Comment by anaskhantts** [comment]
Increase the Git buffer size to the largest individual file size of your repo:
`git config --global http.postBuffer 157286400`

WORKED FOR ME!

**Reply by egylinux** [reply]
thanks, worked for me 👍

**Reply by alaayousf** [reply]
it worked for me.  👍

**Reply by johnny-mnemonica** [reply]
THANK YOU

**Reply by alexandro-us** [reply]
thanks, worked for me!

**Reply by YsaCabrales** [reply]
Thanks! Worked for me <3

**Reply by pappudutta** [reply]
this works for me thanks

**Reply by hegybirog** [reply]
thanks, worked for me!

**Reply by kortiz47** [reply]
thank you - worked for me!

**Reply by khiemdnt** [reply]
great, thanks

**Reply by justege** [reply]
thanks mate

**Reply by dmrd2653** [reply]
thank you :D

**Reply by g-Anant** [reply]
Works like a charm. Thanks!

**Reply by jmurrii-dev** [reply]
thank you 🙏🏻

**Comment by goktugcy** [comment]
Make sure you haven't included node_modules or similar artifacts. Have you tried rolling back to the previous commit and attempting it again?
git reset --hard HEAD

**Comment by EnriquezAlberto** [comment]
Based on the information provided, it appears that the push process might be temporarily delayed due to network issues, server load, or other factors. Aborting the first push and trying again sometimes helps if the initial process got stuck. Since the second attempt also followed similar steps, it is possible that the problem persisted.

Here are some things you might consider doing if you run into a similar situation in the future:

Check network connection – Make sure your network connection is stable and not experiencing interruptions. A slow or unstable network can affect the speed of the boost.

Check the status of the remote server – Sometimes the remote server hosting your Git repository can experience problems or a lot of traffic. You can check the status of the remote server to see if there are any reported issues.

Try again after a while – If the problem persists, you might want to wait for a while and then try again later. Temporary network or server problems may resolve themselves.

Check Git configurations – Make sure your Git configurations are configured correctly, including your username, email, and remote repository URL.

Use SSH – If you're using HTTPS to connect to the remote repository, consider configuring SSH authentication, as it can sometimes provide a more reliable connection.

Contact the repository host – If the problem persists and you are using a hosted Git service (such as GitHub or GitLab), you can contact their support for help.

Remember that Git pushes can take longer for larger repositories or when dealing with larger files. Also, while aborting a send and trying again can sometimes help, it is not a guaranteed solution, and it is important to investigate potential underlying issues for a more permanent resolution.

**Comment by isidorae** [comment]
This issue was solved for me after moving a psd file I had inside my proyect folder (I hadnt done a 'git add' of it or anything, it was just there).

**Comment by UsamaEjaz0** [comment]
I had a an outer git repository initialized. Removing the .git file and correcting it fixed it for me.

**Comment by rozendox** [comment]
when I had this problem, I decided to split the commits, because they were very large. maybe this is the solution

**Reply by MohammedMogeab** [reply]
just bro . make `git gc ` after you have the stuck . it will be done . that 's issue because you made a big comment .
